src/hsl.rs

```diff
@@ -150,15 +167,21 @@ impl<Wp, T> FromColor<Wp, T> for Hsl<Wp, T>
             hue: h.into(),
             saturation: s,
             lightness: l,
-            white_point: PhantomData,
+            space: PhantomData,
         }
     }
 
-    fn from_hsl(hsl: Hsl<Wp, T>) -> Self {
-        hsl
+    fn from_hsl<Sp: RgbSpace<WhitePoint=Wp>>(hsl: Hsl<Sp, T>) -> Self {
+        if TypeId::of::<Sp::Primaries>() == TypeId::of::<S::Primaries>() {
```

Comment:
I am not sure if this is even possible, but can this be implemented using the PartialEq trait
```{rust}

 +impl<S, Wp, T> FromColor<Wp, T> for Hsl<S, T>
     where T: Float,
        S: RgbSpace<WhitePoint=Wp> + PartialEq,
        Wp: WhitePoint,
{
     ...
    fn from_hsl<Sp: RgbSpace<WhitePoint=Wp> + PartialEq>(hsl: Hsl<Sp, T>) -> Self {
    ...
}
```